modules/javafx.controls/src/main/java/javafx/scene/control/skin/TableColumnHeader.java

```diff
@@ -586,8 +590,174 @@ private void doColumnAutoSize(TableColumnBase<?,?> column, int cellsToMeasure) {
 
         // if the prefWidth has been set, we do _not_ autosize columns
         if (prefWidth == DEFAULT_COLUMN_WIDTH) {
-            TableSkinUtils.resizeColumnToFitContent(getTableSkin(), column, cellsToMeasure);
-//            getTableViewSkin().resizeColumnToFitContent(column, cellsToMeasure);
+            resizeColumnToFitContent(column, cellsToMeasure);
+        }
+    }
+
+    /**
+     * Resizes the given column based on the preferred width of all items contained in it. This can be potentially very
+     * expensive if the number of rows is large. Subclass can either call this method or override it (no need to call
```

Comment:
Hmm... I think the doc is a) not entirely correct and b) overspecific on this particular implementation

a) Resizes the column based on content of header _and_ content of cells
b) This implementation measures the preferred width of the header, the preferred width of the first maxRow cells configured with their corresponding content and sizes the column to the max width of all measured values. 

Custom implementations can ... [whatever - exclude headers, use the last n, exclude null content, use min ... ]

Maybe the different aspects should be separated into distinct methods (that's what we did way back in SwingX, those were the days ;) , names pending ...

protected void resizeColumnToFitContent(...) {
     ...
    // measure
    double headerWidth = measureHeaderWidth(...);
    double contentWidth = measureContentWidth(...);
    // do the sizing 
   resizeColumnToFit(headerWidth, contentWidth);

The default implementations can be the same as now, custom descendents can override each aspect separately. In my particular use-case the header was very different from the default, such that the default measure was way off.

Reply:
that's still too specific, IMO: the general contract should be to auto-size _somehow_ based on content of header and cells. This specific implementation uses max of _prefWidth_ of all contributing nodes. There might be others: min/max or header-only or cell-only or first cells in unsorted backing list or not-empty cells ... the contract shouldn't block those :)

As to the additional methods, after sleeping over it: if we have a single override hook now we can add them at a later time, otherwise the scope of this issue might explode.

Reply:
Do you have a Javadoc proposition for this point?